### DISCUSSION TITLE
How-to do camera rotation as a "turntable" effect

### DISCUSSION THREAD

**Discussion by xarthurx** [libigl/libigl#1869] [Q&A]
I've checked every issue related to camera manipulation and am still a bit confused...
And it is related to issue https://github.com/libigl/libigl/issues/1076

There're several parameters:
```
// Camera parameters
  float camera_base_zoom;
  float camera_zoom;
  bool orthographic;
  Eigen::Vector3f camera_base_translation;
  Eigen::Vector3f camera_translation;
  Eigen::Vector3f camera_eye;
  Eigen::Vector3f camera_up;
  Eigen::Vector3f camera_center;
  float camera_view_angle;
```
what is the difference between `zoom` and `base_zoom`, as well as `translation` and `base_translation`?

My goal is to create a "turntable-like" effect so that the model will "rotate" constantly, as a showcase scenario.
What should the correct process be?

Thanks.

**Comment by alecjacobson** [comment]
sigh. the camera parameters in the Viewer began a mess and never really recovered. This effect is certainly possible. @gmin7 was working on ~this~ cleaning up the camera (I think) but I don't remember what the upshot was.

**Reply by gmin7** [reply]
Hi Alec, unfortunately we had not ever gotten to discussing what had to be done for refactoring the viewer camera... but @xarthurx from what I could tell from working on ImGuizmo integration, I believe camera_zoom and translation are set by how the user has manipulated the camera through the GUI, and those are used, in addition to the base zoom and base translation, to compute the transformation matrices applied when rendering the next frame. 

There is likely a much better way to do this, but I would track down where openGL is being passed the view matrix, figure out where the viewer core is storing the user's mouse click events with respect to rotation, then programmatically update that value to achieve the spinning effect.

**Reply by xarthurx** [reply]
OK. It seems I need to spend some effort to make it on my own. As a user, I'd say it will be really helpful and useful to refine this part and have clear documentation. It is sth needed quite often, and will also save your time for answering questions like this.

Thanks.
